### ISSUE TITLE
Codesign: Resource Fork, Finder Information, or Similar Detritus Not Allowed

### ISSUE THREAD

**Issue by akaTiger** [Nuitka/Nuitka#2465]
```
Nuitka: 1.8.2 Installed via pip
Commercial: None
Python: 3.10.5 (v3.10.5:f377153967, Jun  6 2022, 12:36:10) [Clang 13.0.0 (clang-1300.0.29.30)]
Flavor: CPython Official
Executable: /Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10
OS: Darwin (MacOS 12.0.1)
Arch: arm64 (Apple M1)
Version C compiler: /usr/bin/clang (clang).

PyPI (only relevant):
- NumPy
- Tkinter
- Matplotlib
- Sqlite3
```

Command that failed:

```
python3.10 -m nuitka 
--macos-create-app-bundle 
--macos-app-icon=/Users/tiger/Documents/GRTK_version1/grtk/icon.png 
--standalone 
--enable-plugin=tk-inter 
main.py
```

Error:

```
FATAL: Error, call to 'codesign' failed: 
['/usr/bin/codesign', '-s', '-', '--force', '--deep', '--preserve-metadata=entitlements', 'main.app/Contents/MacOS/main',
 'main.app/Contents/MacOS/_asyncio.so', 'main.app/Contents/MacOS/_blake2.so', 'main.app/Contents/MacOS/_bz2.so', 
'main.app/Contents/MacOS/_codecs_cn.so', 'main.app/Contents/MacOS/_codecs_hk.so', 
'main.app/Contents/MacOS/_codecs_iso2022.so', 'main.app/Contents/MacOS/_codecs_jp.so', 
'main.app/Contents/MacOS/_codecs_kr.so', 'main.app/Contents/MacOS/_codecs_tw.so', 
'main.app/Contents/MacOS/_contextvars.so', 'main.app/Contents/MacOS/_csv.so', 'main.app/Contents/MacOS/_ctypes.so', 
'main.app/Contents/MacOS/_datetime.so', 'main.app/Contents/MacOS/_decimal.so', 'main.app/Contents/MacOS/_hashlib.so', 
'main.app/Contents/MacOS/_heapq.so', 'main.app/Contents/MacOS/_lzma.so', 'main.app/Contents/MacOS/_md5.so', 
'main.app/Contents/MacOS/_multibytecodec.so', 'main.app/Contents/MacOS/_multiprocessing.so', 
'main.app/Contents/MacOS/_opcode.so', 'main.app/Contents/MacOS/_pickle.so', 'main.app/Contents/MacOS/_posixshmem.so', 
'main.app/Contents/MacOS/_posixsubprocess.so', 'main.app/Contents/MacOS/_queue.so', 
'main.app/Contents/MacOS/_random.so', 'main.app/Contents/MacOS/_scproxy.so', 'main.app/Contents/MacOS/_sha1.so', 
'main.app/Contents/MacOS/_sha256.so', 'main.app/Contents/MacOS/_sha3.so', 'main.app/Contents/MacOS/_sha512.so', 
'main.app/Contents/MacOS/_socket.so', 'main.app/Contents/MacOS/_sqlite3.so', 'main.app/Contents/MacOS/_ssl.so', 
'main.app/Contents/MacOS/_statistics.so', 'main.app/Contents/MacOS/_struct.so', 'main.app/Contents/MacOS/_tkinter.so', 
'main.app/Contents/MacOS/_uuid.so', 'main.app/Contents/MacOS/array.so', 'main.app/Contents/MacOS/binascii.so', 
'main.app/Contents/MacOS/contourpy/_contourpy.so', 'main.app/Contents/MacOS/fcntl.so', 'main.app/Contents/MacOS/grp.so', 
'main.app/Contents/MacOS/kiwisolver/_cext.so', 'main.app/Contents/MacOS/markupsafe/_speedups.so', 
'main.app/Contents/MacOS/math.so', 'main.app/Contents/MacOS/matplotlib/_c_internal_utils.so', 
'main.app/Contents/MacOS/matplotlib/_image.so', 'main.app/Contents/MacOS/matplotlib/_path.so', 
'main.app/Contents/MacOS/matplotlib/_qhull.so', 'main.app/Contents/MacOS/matplotlib/_tri.so', 
'main.app/Contents/MacOS/matplotlib/backends/_backend_agg.so', 'main.app/Contents/MacOS/matplotlib/backends/_tkagg.so',
 'main.app/Contents/MacOS/matplotlib/ft2font.so', 'main.app/Contents/MacOS/mmap.so', 
'main.app/Contents/MacOS/numpy/core/_multiarray_tests.so', 'main.app/Contents/MacOS/numpy/core/_multiarray_umath.so',
 'main.app/Contents/MacOS/numpy/fft/_pocketfft_internal.so', 'main.app/Contents/MacOS/numpy/linalg/_umath_linalg.so', 
'main.app/Contents/MacOS/numpy/random/_bounded_integers.so', 'main.app/Contents/MacOS/numpy/random/_common.so', 
'main.app/Contents/MacOS/numpy/random/_generator.so', 'main.app/Contents/MacOS/numpy/random/_mt19937.so', 
'main.app/Contents/MacOS/numpy/random/_pcg64.so', 'main.app/Contents/MacOS/numpy/random/_philox.so', 
'main.app/Contents/MacOS/numpy/random/_sfc64.so', 'main.app/Contents/MacOS/numpy/random/bit_generator.so', 
'main.app/Contents/MacOS/numpy/random/mtrand.so', 'main.app/Contents/MacOS/pyexpat.so', 
'main.app/Contents/MacOS/select.so', 'main.app/Contents/MacOS/termios.so', 'main.app/Contents/MacOS/unicodedata.so', 
'main.app/Contents/MacOS/zlib.so', 'main.app/Contents/MacOS/Python', 'main.app/Contents/MacOS/libssl.1.1.dylib', 
'main.app/Contents/MacOS/libcrypto.1.1.dylib', 'main.app/Contents/MacOS/libtcl8.6.dylib', 
'main.app/Contents/MacOS/libtk8.6.dylib', 'main.app/Contents/MacOS/libopenblas64_.0.dylib', 
'main.app/Contents/MacOS/libgfortran.5.dylib', 'main.app/Contents/MacOS/libgcc_s.2.dylib'] -> 
b'main.app/Contents/MacOS/main: resource fork, Finder information, or similar detritus not allowed'.
```

Comments:

Source code can be successfully compiled without `--enable-plugin=tk-inter ` , however, it does nothing when the bundle is opened. The error pops out after the python level compilation and optimization and C files compilation.

Any help would be extremely thankful.

**Comment by kayhayen** [comment]
Yes, tkinter is a problem, and I was expecting to see a duplicate there, mostly because of data files, like in #1568  and this looks like the situation deteriorated further.

**Comment by kayhayen** [comment]
According to this https://developer.apple.com/library/archive/qa/qa1940/_index.html it might be that you viewed the original files in finder, and now they are no longer possible to sign because of that. You are not providing a reproducer, which is bad by itself, so it's impossible to tell if something uploads files to PyPI with that being the case maybe, although extended attributes maybe wouldn't be transported via the zip format of wheels, so that cannot really be it.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by akaTiger** [comment]
```
Nuitka:INFO: Created binary that runs on macOS 11.0 (arm64) or higher.
Nuitka:INFO: Successfully created 'main.app/Contents/MacOS/main'.
```

Thanks for this wonderful rapid fix. 

By switching Nuitika to the factory branch, my Python script could successfully compile and pack into a bundle. For the duplication issue, I figured out that the compilation would go smoothly by deleting the previous build completely (clearing the trash bin). 

But now there appears to be a newer problem: When I finished compiling and tried to open it, it crashed.

To be specific, the verifying process seems to have zero issues. But right after that, the app seemed to crash immediately without anything popping up. The first thing that came to my mind was that there might be some plugins that I haven't imported, so I went back and added `--enable-plugin=numpy` into the command line, but it turns out that the app still crashes under this solution.

I've also tried to check again if there is a problem in my source code. Therefore, I right-clicked on the bundle and went to the "Show Package Contents". Within the "MacOS" folder inside the bundle, I was able to open it as a script and the GUI seems to be working. But it just keeps crashing if I open the whole bundle. (From that I don't think there's any problem with my code)

I've also tried to add the `--enable-console` to figure out what was going on. But the result is disappointing, the console wouldn't even open before the app crashed. I've also realized that some MacOS apps tend to have a popup window during crashes that tells the user what exactly is going on, but unfortunately, there's nothing I can show.

Sorry for the limited information here, there isn't really an error code or logs that I can provide. But as always, any help will be extremely appreciated.

<img width="202" alt="2023-10-05" src="https://github.com/Nuitka/Nuitka/assets/74623731/b70de21f-9ee9-4977-9333-ade3d98aadf3">
